### PR TITLE
Don't require passlib dependency

### PR DESCRIPTION
**Pull request by holmanb**
```
Don't require passlib dependency

Thee crypt module suffices on old versions of Python.
Use it when available. 
```

TODO: tests

### CODE REVIEW

**Comment by TheRealFalcon** [comment]
I know this is still WIP, but we'll want to move `passlib` from `requirements.txt` to `test-requirements.txt`, right?

**Comment by TheRealFalcon** [comment]
Looking at the test failure, looks like the netbsd code never worked on <= 3.6, so looks like the test will need a version gate now.

File: integration-requirements.txt [hunk 1]
```diff
@@ -10,3 +10,4 @@ pycloudlib==1!3.0.2
 pytest<=7.3.1
 
 packaging
+passlib
```
Comment (TheRealFalcon) [inline integration-requirements.txt:13]: Nit: Is this actually needed? AFAIK, integration requirements get installed on the host system, not on the target where tests actually run.

**Review by TheRealFalcon — APPROVED** [review]
Nit inline but LGTM.

File: integration-requirements.txt (hunk 1 above)
Comment (holmanb) [reply integration-requirements.txt:13]: > Is this actually needed?

Yes - `tests/integration_tests/bugs/test_gh671.py` imports it on the host system.

**Pull request merged**